sstables/sstables.hh

```diff
@@ -948,6 +949,14 @@ class file_io_extension {
 public:
     virtual ~file_io_extension() {}
     virtual future<file> wrap_file(sstable&, component_type, file, open_flags flags) = 0;
+    // optionally populate a map of attributes for a given sstable,
+    // suitable for "describe".
+    // This would preferably be interesting info on what/why the extension did
+    // to this table.
+    using attr_value_type = std::variant<sstring, std::map<sstring, sstring>>;
+    using attr_value_map = std::map<sstring, attr_value_type>;
+    virtual void get_attributes(const sstable&, attr_value_map& atts) const
```

Comment:
Why not return the attribute map?

Reply:
Because I want to allow an extension to _add_ either single attributes, or _modify_ them. Obviously, merging of maps can be done on the higher level, but here I opted to add-to-or-replace-in-set functionality direct.

Reply:
Api changed.